mtenn/model.py

```diff
@@ -10,10 +11,65 @@ class Model():
     the data object, but if any operations are required on the data the
     `model_call` function should take care of them.
     """
-    def __init__(self, model, model_call):
+    def __init__(self, representation, strategy):
         super(Model, self).__init__()
-        self.model: torch.nn.Module = model
-        self.model_call: function = model_call
+        self.representation: Representation = representation
+        self.strategy: Strategy = strategy
 
-    def __call__(self, data):
-        return self.model_call(self.model, data)
+    def get_representation(self, *args, **kwargs):
+        """
+        Takes system topolgy and coordinates and returns Nxhidden dimension
+        representation.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        """
+
+        return(self.representation(*args, **kwargs))
+
+    def forward(self, comp, *parts):
+        complex_rep = self.get_representation(*comp)
+        parts_rep = [self.get_representation(*p) for p in parts]
```

Comment:
one thing that I'm a bit unsure on is the argument unpacking here. I have this in because the SchNet model takes the atomic numbers and position vectors as separate arguments, but I don't know if it makes sense to handle that here in the general Model class or if I should push it to `conversion_utils.schnet.get_representation`

Reply:
I think it's safe to assume that atomic numbers and positisions are inputs of any equivariant gnn model

Reply:
yeah definitely, but the e3nn model I've been using takes the input as a single object and extracts the necessary information in its forward call:
https://github.com/e3nn/e3nn/blob/03b40770fa752cfd524321dbb15525e7ccff773b/e3nn/nn/models/gate_points_2101.py#L295

so in order for the code I have here to be compatible with the e3nn model, each data object would have to be passed in as `(data,)` instead of just passing `data`

Reply:
I think the best solution would be to provide wrappers for each model, like

```python

class E3NN(OriginalE3NN):
        def forward(self, x, h):
                data = torch.cat([x, h], dim=-1)
                return super().forward(data)
```

something like that

Reply:
@yuanqing-wang implemented this in the two newest commits, let me know what you think!